jib-core/src/main/java/com/google/cloud/tools/jib/configuration/BuildConfiguration.java

```diff
@@ -115,40 +114,36 @@ public Builder setLayerConfigurations(List<LayerConfiguration> layerConfiguratio
     public BuildConfiguration build() {
       // Validates the parameters.
       List<String> errorMessages = new ArrayList<>();
-      ImageReference baseImageReference = baseImageConfiguration.getImage();
-      ImageReference targetImageReference = targetImageConfiguration.getImage();
-      if (baseImageReference == null) {
-        errorMessages.add("base image is required but not set");
+      if (baseImageConfiguration == null) {
+        errorMessages.add("base image configuration is required but not set");
       }
-      if (targetImageReference == null) {
-        errorMessages.add("target image is required but not set");
+      if (targetImageConfiguration == null) {
+        errorMessages.add("target image configuration is required but not set");
       }
 
       switch (errorMessages.size()) {
         case 0: // No errors
-          if (baseImageReference == null || targetImageReference == null) {
+          if (baseImageConfiguration == null || targetImageConfiguration == null) {
             throw new IllegalStateException("Required fields should not be null");
           }
-          if (baseImageReference.usesDefaultTag()) {
+          if (baseImageConfiguration.getImage().usesDefaultTag()) {
             buildLogger.warn(
                 "Base image '"
-                    + baseImageReference
+                    + baseImageConfiguration.getImage()
                     + "' does not use a specific image digest - build may not be reproducible");
           }
+
+          // If a container configuration hasn't been specified, use an empty one by default
```

Comment:
Can this be nullable too?

Reply:
If we leave it null, we'd need to add extra checks to a bunch of the getters; or worse, if we consolidate the getters into a single `getContainerConfiguration` later, then that's an extra check we'd need to do everywhere we try to get something from the container configuration. And if we instead error on a `ContainerConfiguration` not being passed into the builder like we do for the base/target image configuration, that's just a pain for the user to have to pass in a default `ContainerConfiguration` themselves. It seems better to me to instantiate the container configuration with null/default values if the user hasn't configured it themselves. Plus the worry about constructing an empty `ContainerConfiguration` being potentially expensive is purely hypothetical at this point.

Reply:
Hmm... so this is like a default value? Isn't it customary to set default values when declaring fields?

```java
  private ContainerConfiguration containerConfiguration = ContainerConfiguration.builder().build();
```